src/types.rs

```diff
@@ -133,6 +135,91 @@ pub struct Union {
     pub fields: NamedStructFields,
 }
 
+/// Declaration of an `impl` block.
+///
+/// **Example input:**
+///
+/// ```no_run
+/// # #[cfg(FALSE)]
+/// impl MyType {
+///     // ...
+/// }
+/// ```
+/// or:
+/// ```no_run
+/// # #[cfg(FALSE)]
+/// impl MyTrait for MyType {
+///     // ...
+/// }
+/// ```
+#[derive(Clone, Debug)]
+pub struct Impl {
+    pub attributes: Vec<Attribute>,
+    pub tk_impl: Ident,
+    pub impl_generic_params: Option<GenericParamList>,
+    pub trait_ty: Option<TyExpr>,
+    pub tk_for: Option<Ident>,
+    pub self_ty: TyExpr,
+    pub where_clause: Option<WhereClause>,
+    pub body: ImplBody,
+}
+
+/// The `{ }` group representing the body of an `impl` block.
+#[derive(Clone, Debug)]
+pub struct ImplBody {
+    pub members: Vec<ImplMember>,
+    pub tk_braces: GroupSpan,
+}
```

Comment:
Could probably be inlined in `Impl`? AFAICT `Impl` only ever stores one `ImplBody`, never zero or multiple.